[docs]cluster restart/optimistically confirmed (#31710)
solana-labs/solana: docs/src/terminology.md

A set of [validators](#validator) maintaining a single [ledger](#ledger).

## cluster restart
When a [cluster](#cluster) stops finalizing blocks for a prolonged period of

> time, as determined by social consensus amongst the validator operators, a

Reviewer: `stops finalizing blocks for a prolonged period of time` can be misleading because it' doesn't necessarily have to be a long time. Maybe `due to an unrecoverable set of failures`

Author: Done

## confirmed block

A [block](#block) for which a [super majority](#supermajority) of
stake-weighted [votes](#vote) have been observed. Cannot be rolled

> back given < 5% defective stake.

Reviewer 2: "defective" isn't currently a term we really use for stake, I'm worried this'll just cause different confusion.

Perhaps this: This block cannot be rolled back as long as less than 5% of the current stake weight is dishonest

Reviewer 3: i don't find the terms that over-specify intention to be particularly useful. a buggy implementation is just as problematic as one deliberately designed to do harm in these scenarios. we've used "non-conforming" in the past, but maybe not in docs

## confirmed block

> A [block](#block) that has received [super majority](#supermajority)
> stake-weighted [votes](#vote) should not get rolled back given < 5% malicious.

Reviewer 3: ```suggestion
A [block](#block) for which a [super majority](#supermajority) of
stake-weighted [votes](#vote) have been observed. Cannot be rolled
back given < 5% defective stake.
```

## confirmed block

> See [optimistically confirmed block](#optimistically-confirmed-block).

Reviewer: Do we need this separate section from `optimistically confirmed block`?  Seems like we should only need one and just use the terminology throughout this doc consistently.

The number of [validators](#validator) participating in a [cluster](#cluster).

## optimistically confirmed block

A [block](#block) which got the [ledger votes](#ledger-vote) from
[validators](#validator) holding more than 2/3 of total [stake](#stake) in a

> [cluster](#cluster). The votes may or may not have been confirmed to a block.

Reviewer: Think we can simplify this to: `A block that has received supermajority stake-weighted votes should not get rolled back given < 5% malicious. Details can be found at [Optimistic Confirmation](optimistic_confirmation.md#Primitives)`

Author: Done
